From 1c57964ec7810fc29d511991d4d4b5d56c7c93c2 Mon Sep 17 00:00:00 2001
From: xufuji456 <839789740@qq.com>
Date: Sat, 16 Dec 2023 18:25:46 +0800
Subject: [PATCH] avfilter/metal: fix compile below iOS 13

Here is the msg: "libavfilter/metal/utils.m:35: error: 'MTLGPUFamilyCommon3' is only available on iOS 13.0"
More Info: https://developer.apple.com/documentation/metal/mtlgpufamily/mtlgpufamilycommon3

Signed-off-by: xufuji456 <839789740@qq.com>
---
 libavfilter/metal/utils.m | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavfilter/metal/utils.m b/libavfilter/metal/utils.m
index f365d3ceea..bb1825ae35 100644
--- a/libavfilter/metal/utils.m
+++ b/libavfilter/metal/utils.m
@@ -31,7 +31,7 @@ void ff_metal_compute_encoder_dispatch(id<MTLDevice> device,
     BOOL fallback = YES;
     // MAC_OS_X_VERSION_10_15 is only defined on SDKs new enough to include its functionality (including iOS, tvOS, etc)
 #ifdef MAC_OS_X_VERSION_10_15
-    if (@available(macOS 10.15, iOS 11, tvOS 14.5, *)) {
+    if (@available(macOS 10.15, iOS 13, tvOS 14.5, *)) {
         if ([device supportsFamily:MTLGPUFamilyCommon3]) {
             MTLSize threadsPerGrid = MTLSizeMake(width, height, 1);
             [encoder dispatchThreads:threadsPerGrid threadsPerThreadgroup:threadsPerThreadgroup];
-- 
2.39.3 (Apple Git-145)

